comp/core/workloadmeta/store.go

```diff
@@ -374,7 +374,11 @@ func (w *workloadmeta) GetImage(id string) (*ContainerImageMetadata, error) {
 // Notify implements Store#Notify
 func (w *workloadmeta) Notify(events []CollectorEvent) {
 	if len(events) > 0 {
-		w.eventCh <- events
+		select {
+		case w.eventCh <- events:
+		default:
+			log.Warnf("workloadmeta event channel full, dropping %d events", len(events))
+		}
```

Comment:
Before:
`Notify()` waits until space becomes available in `w.eventCh` buffer.

After:
Drop `events` if `w.eventCh` is full.
We can increase buffer size via `workloadmeta.collector_event_chan_buffer`.